src/GafferSceneUI/VisualiserTool.cpp

```diff
@@ -223,430 +417,1003 @@ class VisualiserGadget : public Gadget
+
+							Imath::V3f worldPos;
+							o2w.multVecMatrix( points[i], worldPos );
+							Imath::V2f rasterPos = viewportGadget->worldToRasterSpace( worldPos );
+							if( rasterBounds.intersects( rasterPos ) )
+							{
+								if( !vData )
+								{
+									auto data = runTimeCast<IntData>( vertexValue );
+									if( !data )
+									{
+										data.reset( new IntData() );
+									}
+									data->writable() = i;
+									vertexValue = data;
+								}
+								else
+								{
+									if( auto iData = runTimeCast<const IntVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<IntData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new IntData() );
+										}
+										data->writable() = iData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto fData = runTimeCast<const FloatVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<FloatData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new FloatData() );
+										}
+										data->writable() = fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto v2fData = runTimeCast<const V2fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<V2fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new V2fData() );
+										}
+										data->writable() = v2fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto v3fData = runTimeCast<const V3fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<V3fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new V3fData() );
+										}
+										data->writable() = v3fData->readable()[i];
+										vertexValue = data;
+									}
+									if( auto c3fData = runTimeCast<const Color3fVectorData>( vData.get() ) )
+									{
+										auto data = runTimeCast<Color3fData>( vertexValue );
+										if( !data )
+										{
+											data.reset( new Color3fData() );
+										}
+										data->writable() = c3fData->readable()[i];
+										vertexValue = data;
+									}
+								}
+
+								// Update cursor value
+								//
+								// NOTE : We defer drawing of the value currently under the cursor, so
+								//        draw the last value label if we replace the cursor value
+
+								const float distance2 = ( cursorRasterPos - rasterPos ).length2();
+								if( ( distance2 < cursorRadius2 ) && ( distance2 < minDistance2 ) )
+								{
+									std::swap( cursorVertexData, vertexValue );
+									std::swap( cursorVertexRasterPos, rasterPos );
+									minDistance2 = distance2;
+								}
+
+								// Draw value label
+
+								if( vertexValue )
+								{
+									const std::string text = stringFromData( vertexValue.get() );
+
+									drawStrokedText(
+										viewportGadget,
+										text,
+										m_tool->sizePlug()->getValue(),
```

Comment:
I suspect it's worth lifting `sizePlug()->getValue()` out of the per-vertex loop.

Reply:
It already was in fact, but far enough up I hadn't made the connection. Fixed in https://github.com/GafferHQ/gaffer/pull/6212/commits/bc7f64a0bbb75d92493abce0f9abb40681d27270